clients/src/main/java/org/apache/kafka/common/security/oauthbearer/internals/secured/HttpAccessTokenRetriever.java

```diff
@@ -371,22 +369,17 @@ static String formatAuthorizationHeader(String clientId, String clientSecret, bo
         return String.format("Basic %s", encoded);
     }
 
-    static String formatRequestBody(String scope) throws IOException {
-        try {
-            StringBuilder requestParameters = new StringBuilder();
-            requestParameters.append("grant_type=client_credentials");
-
-            if (scope != null && !scope.trim().isEmpty()) {
-                scope = scope.trim();
-                String encodedScope = URLEncoder.encode(scope, StandardCharsets.UTF_8.name());
-                requestParameters.append("&scope=").append(encodedScope);
-            }
+    static String formatRequestBody(String scope) {
+        StringBuilder requestParameters = new StringBuilder();
+        requestParameters.append("grant_type=client_credentials");
 
-            return requestParameters.toString();
-        } catch (UnsupportedEncodingException e) {
-            // The world has gone crazy!
-            throw new IOException(String.format("Encoding %s not supported", StandardCharsets.UTF_8.name()));
```

Comment:
Just out of curiosity, all linux distributions aleady support `utf-8`?

Reply:
This exception can be thrown by `URLEncoder.encode(String, String)` when the 2nd argument is not a valid charset. Since Java 10, there's a new method `encode(String, Charset)`. We can now use this new method and pass `StandardCharsets.UTF_8`, which is a `Charset` object directly instead of `StandardCharsets.UTF_8.name` which is a `String`. That way we are sure `encode(String, Charset)` won't throw `UnsupportedEncodingException`.

Reply:
Thanks for your explanation!